[inductor] Register an op for mm_plus_mm

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #104835

Summary: Currently the aten version of mm_plus_mm has no cpp
implementation, and thus cpp_wrapper can not generate the correct cpp
function call for it.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8

Differential Revision: [D47372057](https://our.internmc.facebook.com/intern/diff/D47372057)

desertfire committed:
[inductor] Register an op for mm_plus_mm

Summary: Currently the aten version of mm_plus_mm has no cpp
implementation, and thus cpp_wrapper can not generate the correct cpp
function call for it.

[ghstack-poisoned]

lezcano commented on torch/csrc/inductor/inductor_ops.cpp:
```diff
@@ -0,0 +1,27 @@
+    const Tensor& c,
+    const Tensor& d,
+    Tensor& out) {
+  at::matmul_out(out, a, b);
+  out.addmm_(c, d);
```
Why are you using matmul for one and mm (or addmm) for the other one? We should use either mm for both or matmul for both. I think using `mm` for both would be best here.

desertfire committed:
Update on "[inductor] Register an op for mm_plus_mm"


Summary: Currently the aten version of mm_plus_mm has no cpp
implementation, and thus cpp_wrapper can not generate the correct cpp
function call for it.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D47372057).

desertfire committed:
Update on "[inductor] Register an op for mm_plus_mm"


Summary: Currently the aten version of mm_plus_mm has no cpp
implementation, and thus cpp_wrapper can not generate the correct cpp
function call for it.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8

Differential Revision: [D47372057](https://our.internmc.facebook.com/intern/diff/D47372057)

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D47372057).

desertfire commented:
@pytorchbot merge

clee2000 commented:
@pytorchbot revert -m "breaking internal builds" -c ghfirst

clee2000 commented:
@pytorchbot revert -m "breaking internal builds" -c ghfirst

huydhn commented:
@pytorchbot successfully started a revert job. Check the current status [here](None).
Questions? Feedback? Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)

huydhn commented:
@pytorchbot successfully started a revert job. Check the current status [here](None).
Questions? Feedback? Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)

huydhn commented:
## Reverting PR 104835 failed
**Reason**: module 'os' has no attribute 'exit'

huydhn commented:
@pytorchbot successfully started a revert job. Check the current status [here](None).
Questions? Feedback? Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)

huydhn commented:
## Reverting PR 104835 failed
**Reason**: Object of type dict_keys is not JSON serializable

huydhn commented:
@pytorchbot successfully started a revert job. Check the current status [here](None).
Questions? Feedback? Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)

huydhn commented:
## Reverting PR 104835 failed
**Reason**: module 'os' has no attribute 'exit'

huydhn commented:
@pytorchbot successfully started a revert job. Check the current status [here](None).
Questions? Feedback? Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)

huydhn commented:
## Reverting PR 104835 failed
**Reason**: module 'os' has no attribute 'exit'

huydhn commented:
@pytorchbot successfully started a revert job. Check the current status [here](None).
Questions? Feedback? Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)

huydhn commented:
## Reverting PR 104835 failed
**Reason**: module 'os' has no attribute 'exit'

huydhn commented:
@pytorchbot successfully started a revert job. Check the current status [here](None).
Questions? Feedback? Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)

huydhn commented:
## Reverting PR 104835 failed
**Reason**: module 'os' has no attribute 'exit'

huydhn commented:
@pytorchbot successfully started a revert job. Check the current status [here](None).
Questions? Feedback? Please reach out to the [PyTorch DevX Team](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)
